src/main/java/org/nibor/autolink/internal/WwwUrlScanner.java

```diff
@@ -0,0 +1,42 @@
+package org.nibor.autolink.internal;
+
+import org.nibor.autolink.LinkSpan;
+import org.nibor.autolink.LinkType;
+
+/**
+ * Scan for URLs starting from the trigger character "w", requires "www.".
+ * <p>
+ * Based on RFC 3986.
+ */
+public class WwwUrlScanner extends UrlScanner {
```

Comment:
- [ ] Instead of inheriting from UrlScanner, extract the code of `findLast` into a new utility function, maybe `Scanners.findLastForUrl`.